### PR TITLE
feat: implement validation functions for pricing model diffs

### PR DESCRIPTION
**Pull request by agreea**
## What Does this PR Do?

Implements PR 4 from the billing portal tools gameplan: comprehensive validation functions for pricing model diffs using Zod parsing with strict mode.

### Key Functions Implemented

- **computeUpdateObject**: Identifies fields that differ between existing and proposed objects
- **validateUsageMeterDiff**: Prevents removal of usage meters and validates mutable fields
- **validateFeatureDiff**: Prevents feature type changes and validates mutable fields
- **validatePriceChange**: Prevents price type changes and validates mutable fields
- **validateProductDiff**: Validates product and price updates with strict mode

Includes 66 comprehensive tests covering valid updates, immutable field protections, and edge cases.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Adds strict validation for pricing model diffs and product diffing to prevent unsafe changes. Ensures only allowed fields are updated and surfaces clear errors for immutable fields.

- **New Features**
  - diffProducts returns toRemove/toCreate/toUpdate with optional priceDiff when prices differ.
  - computeUpdateObject returns only changed fields between objects.
  - validateUsageMeterDiff: disallows removals; validates updates via usageMetersClientUpdateSchema (strict).
  - validateFeatureDiff: blocks type changes; validates mutable fields (toggle/usage_credit_grant); allows usageMeterSlug updates.
  - validatePriceChange: blocks type changes and immutable updates across subscription/single_payment/usage price schemas.
  - validateProductDiff: validates product fields and delegates price checks per type.
  - Expanded tests covering valid updates, immutable guards, and edge cases.

- **Migration**
  - Do not remove usage meters; updates only.
  - To change feature or price type, remove and create a new one.
  - Immutable price fields (e.g., unitPrice, intervalUnit/count, usageMeterSlug for usage) now error; create a new price instead.

<sup>Written for commit c02ebae88ee7a53afd35dc466398b78cc4a49119. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * New exported utilities to compute precise field-level updates and validate pricing/product diffs, including stricter rules for immutable fields and disallowed removals/type changes.
* **Tests**
  * Expanded unit tests and helpers covering create/update/remove flows, nested diffs, price-type scenarios, and extensive validation cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/utils/pricingModels/diffing.ts [hunk 1]
```diff
@@ -343,3 +356,303 @@ export const diffProducts = (
...
+    ...Object.keys(proposed),
+  ])
+
+  for (const key of allKeys) {
+    const existingValue = existing[key]
+    const proposedValue = proposed[key]
+
+    // Compare values using JSON stringify for deep equality
+    if (
+      JSON.stringify(existingValue) !== JSON.stringify(proposedValue)
```
Comment (angihe93) [inline platform/flowglad-next/src/utils/pricingModels/diffing.ts]: check for the robustness of this, may need to order the keys first before comparing or perhaps use isEqual from lodash-es
